.github/workflows/test.yml

```diff
@@ -19,40 +18,44 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        # pytest-split automatically distributes work load so parallel jobs finish in similar time
         os: [ubuntu-latest]
         python-version: ["3.11"]
 
     runs-on: ${{ matrix.os }}
 
     env:
       PMG_MAPI_KEY: ${{ secrets.PMG_MAPI_KEY }}
-      PMG_VASP_PSP_DIR: /home/runner/work/matgenb/matgenb/psp
+      PMG_VASP_PSP_DIR: "${{ github.workspace }}/psp"
+      MPLBACKEND: Agg  # non-interactive backend for matplotlib
 
     steps:
       - name: Check out repo
         uses: actions/checkout@v4
 
       - name: Set up micromamba
         uses: mamba-org/setup-micromamba@main
-
-      - name: Create mamba environment
-        run: |
-          micromamba create -n venv python=${{ matrix.python-version }} --yes
-          micromamba install -n venv -c conda-forge enumlib packmol bader openbabel openff-toolkit --yes
+        with:
+          environment-name: matgenb
+          condarc: |
+            channels:
+              - conda-forge
+          create-args: >-
+            python=${{ matrix.python-version }}
+            enumlib
+            packmol
+            bader
+            openbabel
+            openff-toolkit
 
       - name: Install uv
-        run: micromamba run -n venv pip install uv
+        uses: astral-sh/setup-uv@v4
+        with:
+          version: "latest"
 
       - name: Install dependencies
         run: |
-          micromamba activate venv
-          uv pip install numpy cython
-          uv pip install --upgrade -r requirements-ci.txt
+          micromamba run -n matgenb uv pip install --upgrade -r requirements-ci.txt
 
-      - name: pytest
+      - name: Run pytest
         run: |
-          micromamba activate venv
-          cd notebooks
-          pwd
-          pytest --ignore-glob=*notest.ipynb --nbmake .
```

Comment:
TODO: either use `notest` pattern to ignore file, or insert `pytest.skip` mark (not sure how to achieve this for ipynb though)